hw9: advanced oop/hw9.py

```diff
@@ -0,0 +1,212 @@
+#####################################################################################
+# this homework explores the functionality of object-oriented programming in Python #
+#####################################################################################
+
+# task 1: connected classes Chat, Message and User
+
+import datetime
+
+
+class Chat:
+    def __init__(self):
+        self.chat_history = []
+
+    def show_last_message(self):
+        # the assumption is made that we live in the one timeline
+        # therefore, the last message is one sent as the latest
+        last_message = self.chat_history[0]
+        return last_message.show()
+
+    def get_history_from_time_period(self, start, stop):
+        chat_sorted = Chat()  # type(self) ?
+
+        for message in self.chat_history[::-1]:
+            if start <= message.datetime <= stop:
+                chat_sorted.recieve(message)
+
+        return chat_sorted
+
+    def show_chat(self):
+        for message in self.chat_history:
+            message.show()
+            print()
+
+    def recieve(self, message):
+        self.chat_history = [message] + self.chat_history
+
+
+class Message:
+    def __init__(self, text, date_time, user, ability='is a god/ess'):
+        self.text = text
+        self.datetime = datetime.datetime.strptime(date_time, "%d/%m/%y %H:%M")
+        self.user = User(ability=ability, name=user)
+
+    def show(self):
+        print(self.datetime, self.user.name, self.text)
+
+    def send(self, chat):
+        chat.recieve(self)
```

Comment:
А вот тут уже можно задать значение для `datetime`

```suggestion
    def send(self, chat):
        self.datetime = datetime.datetime.now()
        chat.recieve(self)
```

Reply:
Подход с отправкой сообщения хороший 👍